sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -158,15 +332,21 @@ static ProcessElementMethod create(
     }
 
     /** Whether this {@link DoFn} uses a Single Window. */
-    public boolean usesSingleWindow() {
-      return extraParameters().contains(Parameter.BOUNDED_WINDOW);
+    public boolean ignoresWindow() {
+      for (Parameter p : extraParameters()) {
+        if (p instanceof BoundedWindowParameter
+            || p instanceof StateParameter) {
```

Comment:
Ah, I had done it with an anonymous `Predicate`, which was bulky, so I reverted. Didn't notice the nice combinators. Done.